core/embed/extmod/modtrezorcrypto/modtrezorcrypto-secp256k1_zkp.h

```diff
@@ -325,13 +331,265 @@ STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_multiply(
+  if (MP_OBJ_IS_SMALL_INT(arg)) {
+    return MP_OBJ_SMALL_INT_VALUE(arg);
+  } else if (MP_OBJ_IS_TYPE(arg, &mp_type_int)) {
+    byte buff[8];
+    uint64_t res = 0;
+    mp_obj_t *o = MP_OBJ_TO_PTR(arg);
+
+#if MICROPY_LONGINT_IMPL != MICROPY_LONGINT_IMPL_MPZ
+#error "MPZ supported only"
+#endif
+    mp_obj_int_to_bytes_impl(o, true, 8, buff);
+    for (int i = 0; i < 8; i++) {
+      res <<= i > 0 ? 8 : 0;
+      res |= (uint64_t)(buff[i] & 0xff);
+    }
+    return res;
+  } else {
+    if (MICROPY_ERROR_REPORTING == MICROPY_ERROR_REPORTING_TERSE) {
+      mp_raise_TypeError("can't convert to int");
+    } else {
+      nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_TypeError,
+                                              "can't convert %s to int",
+                                              mp_obj_get_type_str(arg)));
+    }
+  }
+}
+
+/// def pedersen_commit(self, value: long, blinding_factor: bytes, gen: bytes)
+/// -> bytes:
+///     '''
+///     Commit to specified integer value, using given 32-byte blinding factor.
+///     '''
+STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_pedersen_commit(
+    size_t n_args, const mp_obj_t *args) {
+  const secp256k1_context *ctx =
+      mod_trezorcrypto_get_secp256k1_context(args[0]);
+
+  const uint64_t value = _mp_obj_get_uint64(args[1]);
+
+  mp_buffer_info_t blind;
+  mp_get_buffer_raise(args[2], &blind, MP_BUFFER_READ);
+  if (blind.len != 32) {
+    mp_raise_ValueError("Invalid length of blinding factor");
+  }
+  mp_buffer_info_t gen;
+  mp_get_buffer_raise(args[3], &gen, MP_BUFFER_READ);
+  if (gen.len != 33) {
+    mp_raise_ValueError("Invalid length of generator");
+  }
+  secp256k1_generator generator;
+  if (!secp256k1_generator_parse(ctx, &generator, gen.buf)) {
+    mp_raise_ValueError("Generator parsing failed");
+  }
+
+  secp256k1_pedersen_commitment commit;
+  if (!secp256k1_pedersen_commit(ctx, &commit, blind.buf, value, &generator)) {
+    mp_raise_ValueError("Pedersen commit failed");
+  }
+
+  byte output[33] = {0};
+  secp256k1_pedersen_commitment_serialize(ctx, output, &commit);
+  return mp_obj_new_bytes(output, sizeof(output));
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(
+    mod_trezorcrypto_secp256k1_context_pedersen_commit_obj, 4, 4,
+    mod_trezorcrypto_secp256k1_context_pedersen_commit);
 
-//////////////////////////////////////////////////////////////////////////////
+const size_t RANGEPROOF_SIGN_BUFFER_SIZE = 5134;
+
+/// def rangeproof_sign(self, value: int, commit: bytes, blind: bytes,
+///                     nonce: bytes, message: bytes, extra_commit: bytes,
+///                     gen: bytes, scratch_buffer: bytearray) -> memoryview:
+///     '''
+///     Return a range proof for specified value (as a memoryview).
+///     '''
+STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_rangeproof_sign(
+    size_t n_args, const mp_obj_t *args) {
+  // TODO: not sure about the constants values here.
```

Comment:
Hm, yep, I'm not sure either, see
https://github.com/ElementsProject/elements/blob/7b302eb99d2d1bf861429d6bd209771f634b8e8e/src/blind.cpp#L184
 @apoelstra @gsansders  Can you comment?

Reply:
@instagibbs

Reply:
I changed the default `BITS` value to 36.

Reply:
926abd29813bbd4617c5117a689abdf92e9b2139 allows setting the range-proof configuration from MicroPython.